src/shadowbox/infrastructure/prometheus_scraper.ts

```diff
@@ -74,44 +74,41 @@ export function runPrometheusScraper(
     prometheusEndpoint: string): Promise<child_process.ChildProcess> {
   mkdirp.sync(path.dirname(configFilename));
   const ymlTxt = jsyaml.safeDump(configJson, {'sortKeys': true});
-  return new Promise((resolve, reject) => {
-    fs.writeFile(configFilename, ymlTxt, 'utf-8', async (err) => {
-      if (err) {
-        reject(err);
-      }
-      const commandArguments = ['--config.file', configFilename];
-      commandArguments.push(...args);
-      const runProcess = child_process.spawn('/root/shadowbox/bin/prometheus', commandArguments);
-      runProcess.on('error', (error) => {
-        logging.error(`Error spawning prometheus: ${error}`);
-      });
-      // TODO(fortuna): Add restart logic.
-      runProcess.on('exit', (code, signal) => {
-        logging.info(`prometheus has exited with error. Code: ${code}, Signal: ${signal}`);
-      });
-      // TODO(fortuna): Disable this for production.
-      // TODO(fortuna): Consider saving the output and expose it through the manager service.
-      runProcess.stdout.pipe(process.stdout);
-      runProcess.stderr.pipe(process.stderr);
-
-      await waitForPrometheusReady(prometheusEndpoint);
-      resolve(runProcess);
+  fs.writeFileSync(configFilename, ymlTxt, 'utf-8');
```

Comment:
Please revert to the async write. The problem with this one is that it will block the node thread. It's not equivalent to the await with promise, since await doesn't block the thread.

Reply:
Done, added a comment to explain this behavior.